### PR TITLE
Fixes for 'Using perimeter (curved)' labels on polygon layers

### PR DESCRIPTION
**Pull request by fritsvanveen**

### CODE REVIEW

**Comment by fritsvanveen** [comment]
There is a problem with curved labels on polylines. Must fix this first.

**Comment by nyalldawson** [comment]
@fritsvanveen I'd suggest holding off for a little - I've got a commit incoming which reworks some parts of this too.

**Comment by fritsvanveen** [comment]
@nyalldawson No problem

**Comment by nyalldawson** [comment]
@fritsvanveen ok, all done. I've nothing else planned re labeling, so won't be causing merge conflicts again :)

**Comment by fritsvanveen** [comment]
@nyalldawson I have astyled all my committed files (to the best of my knowledge), but Travis still failed me last time. I hope for the best. prepare-commit.sh doesn't work. It can't find astyle.sh for some reason.
Can you test these changes as the code is pretty complex and the permutations to test are extensive. I have converted a polygon to a polyline, so I could test the same looking polyline and polygon. I tested for curved and straight, above and on and below, rotated allowed or not, line orientation checked and unchecked and various repeat settings. I believe everything is working fine now, but I am not overly confident.

**Comment by jef-n** [comment]
It doesn't fail for indentation anymore.   Follow the link to the test results (http://tinyurl.com/hd7c22u).  The labels apparently changed sides and there a two new not doxymented methods.

File: src/core/pal/feature.cpp [hunk 1]
```diff
@@ -1231,99 +1209,121 @@ int FeaturePart::createCurvedCandidatesAlongLine( QList< LabelPosition* >& lPos,
     flags = FLAG_ON_LINE; // default flag
   // placements may need to be reversed if using line position dependent orientation
   // and the line has right-to-left direction
-  bool reversed = ( !( flags & FLAG_MAP_ORIENTATION ) ? isRightToLeft : false );
+  bool reversed = (( flags & FLAG_MAP_ORIENTATION ) ? isRightToLeft : false );
```
Comment (nyalldawson) [inline src/core/pal/feature.cpp]: I think it's this change which is causing the incorrectly flipped labels. I was also very confused by this flag (it seems backwards to me too!),but it must be getting misinterpreted when being used later.

**Comment by fritsvanveen** [comment]
I found the problem. It is with curved placement (line as well as polygon layer) with 'Line orientation dependent position' checked and 'Show upside-down labels' set to 'never'. I did not arrive to this conclusion from the Travis output. Can someone explain this to me?

**Comment by fritsvanveen** [comment]
@nyalldawson The same statement (reversed =...) is present in two more places in the same file (without the negation). So I think the culprit is elsewhere. I don't pretend however to fully grasp the code ;-(. Any insight is welcome.

**Comment by jef-n** [comment]
@fritsvanveen doesn't http://dash.orfeo-toolbox.org/testDetails.php?test=44418702&build=240054 show that problem?

**Comment by fritsvanveen** [comment]
@jef-n Yes, but how do I conclude which user settings were used to arrive at that image.

**Comment by nyalldawson** [comment]
Manually squashed and merged as 34e2bea

**Pull request closed without merging**